internal/benchrunner/runners/common/scenario.go

```diff
@@ -0,0 +1,119 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package common
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/elastic/go-ucfg/yaml"
+)
+
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+const DevPath = "_dev/benchmark/rally"
+
+type Scenario struct {
+	Package     string     `config:"package" json:"package"`
+	Description string     `config:"description" json:"description"`
+	Version     string     `config:"version" json:"version"`
+	DataStream  DataStream `config:"data_stream" json:"data_stream"`
+	Corpora     Corpora    `config:"corpora" json:"corpora"`
+}
+
+type DataStream struct {
+	Name string `config:"name" json:"name"`
+}
+
+type Corpora struct {
+	Generator *Generator `config:"generator" json:"generator"`
+}
+
+type Generator struct {
+	TotalEvents uint64          `config:"total_events" json:"total_events"`
+	Template    CorporaTemplate `config:"template" json:"template"`
+	Config      CorporaAsset    `config:"config" json:"config"`
+	Fields      CorporaAsset    `config:"fields" json:"fields"`
+}
+
+type CorporaAsset struct {
+	Raw  map[string]interface{} `config:"raw" json:"raw"`
+	Path string                 `config:"path" json:"path"`
+}
+type CorporaTemplate struct {
+	Raw  string `config:"raw" json:"raw"`
+	Path string `config:"path" json:"path"`
+	Type string `config:"type" json:"type"`
+}
+
+func DefaultConfig() *Scenario {
+	return &Scenario{}
+}
+
+func ReadConfig(path, scenario, packageName, packageVersion string) (*Scenario, error) {
+	configPath := filepath.Join(path, DevPath, fmt.Sprintf("%s.yml", scenario))
+	c := DefaultConfig()
+	cfg, err := yaml.NewConfigWithFile(configPath)
+	if err != nil {
+		return nil, fmt.Errorf("can't load benchmark configuration: %s: %w", configPath, err)
+	}
+
+	if err == nil {
+		if err := cfg.Unpack(c); err != nil {
+			return nil, fmt.Errorf("can't unpack benchmark configuration: %s: %w", configPath, err)
+		}
+	}
+
+	c.Package = packageName
+	c.Version = packageVersion
+
+	if c.DataStream.Name == "" {
+		return nil, errors.New("can't read data stream name from benchmark configuration: empty")
+	}
+
+	return c, nil
+}
+
+func ReadScenarios(path, scenarioName, packageName, packageVersion string) (map[string]*Scenario, error) {
```

Comment:
Nit. Do we have any benefit on returning scenarios as a map? Returning a list would help controlling the order of execution and results. `Scenario` could have a field for the name.
```suggestion
func ReadScenarios(path, scenarioName, packageName, packageVersion string) ([]*Scenario, error) {
```

Reply:
@jsoriano I would prefer to delay code changes to a different PR. The reason is for this PR I only moved code around, not type / logic change to make sure the refactoring does not affect any of the logic.